File: CUValles2019B/Basico/Problema2/Problema2.py
"""
Problema 2: Iniciales en Minusculas
Autor de la solucion: Luis Arturo DLG.
Python 3.x

Descripcion del problema:Recibiremos una frase, la cual las iniciales de
cada palabra deben ser convertidas en minusculas, y el resto de la palabra en
mayusculas

Entrada: The dArK SidE oF tHe MoOn
Salida: tHE dARK sIDE oF tHE mOON

Explicacion: Debemos de tomar toda la frase (incluyendo espacios en blanco), e
ir cambiando a minusculas a las letras que inician la palabra, esto se puede hacer
buscando los espacios en blanco, una vez que encontramos la inicial, convertimos
a toda letra a mayusculas, esto hasta encontrar otro espacio y repetimos la accion
para toda la frase.
"""

if __name__ == '__main__': # main del programa
    frase = input() # leer la frase
    """Tendremos una variable de tipo booleano a manera de bandera para indicar
    si la letra que vamos a convertir es inicial o no, esta condicion siempre inicia
    en True, una vez encontrado la inicial se cambia a False, hasta encontrar un espacio"""
    inicial = True

    # Modo manual
    for letra in frase:

        if inicial: # si encontramos la inicial
            print(letra.lower(), end="") # el metodo lower convierte en minuscula la letra
            inicial = False # cambiamos la bandera a False

        else: # si no es la inicial
            print(letra.upper(), end="") # el metodo upper convierte en mayuscula a la letra

        if letra.isspace(): # el metodo isspace devuele True si el caracter es un espacio
            inicial = True # cambiamos la bandera a True

    print() # colocamos un print vacio para tener un salto de linea

    """Modo pythonico
    print(frase.title().swapcase())"""

File: CUValles2019B/Basico/Problema2/Problema2.java
/*
Problema 2: Iniciales en Minusculas
Autor de la solucion: AlbertAvile.
Java 8
Descripcion del problema:Recibiremos una frase, la cual las iniciales de
cada palabra deben ser convertidas en minusculas, y el resto de la palabra en
mayusculas
Entrada: The dArK SidE oF tHe MoOn
Salida: tHE dARK sIDE oF tHE mOON
Explicacion: Debemos de tomar toda la frase (incluyendo espacios en blanco), e
ir cambiando a minusculas a las letras que inician la palabra, esto se puede hacer
buscando los espacios en blanco, una vez que encontramos la inicial, convertimos
a toda letra a mayusculas, esto hasta encontrar otro espacio y repetimos la accion
para toda la frase.
*/

import java.util.Scanner;

public class Problema2{
  public static void main(String[] args) {
    Scanner entrada = new Scanner(System.in);
    String cadena = entrada.nextLine();//leemos la cadena

    Boolean inicial = true;//creamos una variable bandera para indicar si la letra es la inicial de cada palabra
    
    for(int i = 0; i < cadena.length(); i++){
      if(inicial){//si la letra es la inicial la convertimos a minuscula
        System.out.print((""+cadena.charAt(i)).toLowerCase());
        inicial = false;//igualamos a false la bandera ya que las siguientes no seran la inicial
      }
      else{
        System.out.print((""+cadena.charAt(i)).toUpperCase());//convertimos las letras no iniciales a mayusculas

        if(Character.isSpace(cadena.charAt(i))){//si encontramos un espacio regresamos la variable a true
            inicial = true;
        }
      }
      
    }
  }
}